### PR TITLE
several gameobj fixes (add boon creatures and friends)

### PR DESCRIPTION
**Pull request by FarFigNewGut**
- adds new boon creature type
  - copied all aggressive npc list instead of trying to figure out which are currently available to boon
  - added new "prefix_required" type table kind to make the prefix field required for match
  - associated script changes to handle new field type, tested generating xml everything seems to work as expected...
  - except "ghostly" (through necrotic ring or boon) don't register as undead for normally not-undead, not sure how to handle that currently.

- addresses #117  (adds ancient (dangerous creature) prefix for monstrosities to SoS migration

- adds tradebars to valuable type / gemshop sellable lists. (#132)

### CODE REVIEW

File: type_data/tables/aggressive_npc.yaml [hunk 1]
```diff
@@ -7,6 +7,67 @@ prefix:
   - veteran
   # confluence
   - huge
+  # boon / glam / boss creatures
```
Comment (ondreian) [inline type_data/tables/aggressive_npc.yaml:10]: These should exist in a migration file. the `.yaml` files are only here to serve as a baseline of the initial GameObj import into the ruby DSL

**Comment by FarFigNewGut** [comment]
Pulled down @sandersch 's changes he sent to move it over to a table migration(s)

**Comment by mrhoribu** [comment]
Looks decent but will need @ondreian and @sandersch to double check.

**Comment by ondreian** [comment]
The code looks ok, I'm concerned there are no additional test cases added to verify the outputs since this is such a big change.  I'll see about pulling it down and building the file to run IG for a bit sometime this week

File: type_data/migrations/30_tradebars.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+  insert(:name, %{small copper tradebar})
+  insert(:name, %{small bronze tradebar})
+  insert(:name, %{small silver tradebar})
+  insert(:name, %{small gold tradebar})
+  insert(:name, %{small platinum tradebar})
+  insert(:name, %{large copper tradebar})
+  insert(:name, %{large bronze tradebar})
+  insert(:name, %{large silver tradebar})
+  insert(:name, %{large gold tradebar})
+  insert(:name, %{large platinum tradebar})
```
[the remark this answers is not in the document]
Comment (FarFigNewGut) [reply type_data/migrations/30_tradebars.rb:11]: I checked my logs going back a several years, there's were the only permutations I found but (those 4 metals, and small / large being optional). but they are super rare it's really only handful found between a combined, i dunno 40m of exp or something or more between various characters. If anyone has any others I can toss them on the pile here also.

File: type_data/migrations/7_sos.rb [hunk 3]
```diff
@@ -8,6 +8,7 @@
 migrate :aggressive_npc, :undead do
   insert(:name, %[shambling lurk])
   insert(:name, %[patchwork flesh monstrosity])
+  insert(:name, %[ancient flesh monstrosity])
```
[the remark this answers is not in the document]
Comment (FarFigNewGut) [reply type_data/migrations/7_sos.rb:11]: It's not possible to get a boon + grizzled npc. I didn't think it was and Tysong pinged ixix and got that confirmed directly pretty sure.

[the remark this answers is not in the document]
Comment (sandersch) [reply type_data/migrations/7_sos.rb:11]: I mean that the grizzled name is irregular so it is likely that the boon name will be irregular as well. ie the boon regex in this PR will look for `glistening patchwork flesh monstrosity` or `scintillating ancient flesh monstrosity` but the boon system would probably name it as `slime covered flesh monstrosity`. Not a blocker at all for merging this

**Comment by sandersch** [comment]
I merged this commits plus some updated specs in #266 (I couldn't update this branch since I don't have push access to your repo). Thanks for adding the boon creatures!

**Pull request closed without merging**